libraries/AP_Relay/AP_Relay.cpp

```diff
@@ -140,6 +141,18 @@ void AP_Relay::set(const uint8_t instance, const bool value)
     if (_pin[instance] == -1) {
         return;
     }
+    const uint32_t now = AP_HAL::millis();
+    _pin_states = value ? _pin_states | (1U<<instance) : _pin_states & ~(1U<<instance);
+    if ((_pin_states != _last_logged_pin_states) && (now > (_last_log_ms + 100))) {
```

Comment:
```suggestion
    if ((_pin_states != _last_logged_pin_states) && (now - _last_log_ms > 100))) {
```

Adding times and intervals leads to bugs at wrap time.  While it's not critical here, having the pattern anywhere in our code is bad as it may get copy/pasted where it will cause real trouble.

Reply:
thanks...will change